src/lib/rest/restServiceLookup.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+*
+* Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include "common/string.h"
+
+#include "rest/ConnectionInfo.h"
+#include "rest/RestService.h"
+#include "rest/restServiceLookup.h"
+
+
+
+/* ****************************************************************************
+*
+* restServiceLookup -
+*/
+RestService* restServiceLookup(ConnectionInfo* ciP, bool* badVerbP)
```

Comment:
Looking to the body of the funtion, only apiVersion, verb and url are used to do the lookup. 

```
RestService* restServiceLookup(Verb verb, ApiVersion apiVersion, const std::string& url, bool* badVerbP)
```

That way it would be more explicit on which exact input data the lookup is done, without needed to look how the function is implemented.

Reply:
This is "your way of doing things" vs "my way of doing things".
Both ways have pros and cons.
Needless to say, I prefer my way.
Especially for one reason. I plan to make the ConnectionInfo a thread variable, to avoid to allocate it for each incoming request. With `ciP->` I will very easily find all occurrences and change to `ci.` (`ci` will become a global thread variable and we not only avoid to allocate it, we also avoid to send it to hundreds of functions).

Reply:
Good enough. I tend to prefer signatures with clear inputs and output parameters (the 101 of modular programming ;) but it is ok keeping the current style. The change should be global and can be defered for a future discussion.

NTC

Reply:
Especially we should discuss about making "ciP" a thread var "ci". Or even, Remove the struct ci and use 50 (or so) thread vars.   Lots of time will be saved by not having to send these connection variables, ideal for global vars, over the stack hundreds of thousands of times every second.

Reply:
Maybe some point in the middle. 

I mean, maybe not all the current member of ConnectionInfo make sense as stanalone variables and they can be grouped into coherent structs. But having a single variable (current ConnectionInfo object) from everything is too much.

However, this is another piece of work. In the context of the PR ye olde ConnectionInfo :) is ok